src/v/cluster/cluster_discovery.cc

```diff
@@ -185,4 +366,37 @@ cluster_discovery::get_node_index_in_seed_servers() {
       boost::numeric_cast<int32_t>(std::distance(seed_servers.cbegin(), it))};
 }
 
+bool cluster_discovery::is_amnesiac_seed_server() const {
+    vassert(
+      _cluster_bootstrap_info_fetched,
+      "Cluster bootstrap info not fetched yet");
+
+    bool failed = false;
+    std::optional<model::cluster_uuid> peer_cluster_uuid;
+    for (auto& peer : _cluster_bootstrap_info) {
```

Comment:
nit: could use structured binding so it's easier to follow instead of `first` and `second`